### PR TITLE
Fixed possible security issue where HTML tags weren't removed

### PR DESCRIPTION
**Pull request by NiLSPACE**
Added a message in the webadmin when the core is loaded

### CODE REVIEW

File: web_chat.lua [hunk 1]
```diff
@@ -260,13 +260,17 @@ end
...
 --- It does this by selecting all the characters between "http(s)://" and a space, and puts an anker tag around it.
-local function CheckForLinks(a_Message)
+local function ParseMessage(a_Message)
 	local function PlaceString(a_Url)
 		return '<a href="' .. a_Url .. '" target="_blank">' .. a_Url .. '</a>'
 	end
-	return a_Message:gsub('http://[^%s]+', PlaceString):gsub('https://[^%s]+', PlaceString)
+	
+	a_Message = a_Message:gsub("<", "&lt;"):gsub(">", "&gt;")
+	a_Message = a_Message:gsub('http://[^%s]+', PlaceString):gsub('https://[^%s]+', PlaceString)
```
Comment (worktycho) [inline web_chat.lua]: This is dangorous. What about the url: `http://example.com"onmouseover="alert('Injection')` The html it produces (`<a href="' http://example.com"onmouseover="alert('Injection')'" target="_blank">`) isn't tencnacally valid but at least firefox accepts it.

Comment (NiLSPACE) [inline web_chat.lua]: Would `gsub("\"", "\\\"")` prevent this?

Comment (worktycho) [inline web_chat.lua]: `http://example.com\"onmouseover="alert('Injection')` gets escaped to `<a href="' http://example.com\\"onmouseover="alert('Injection')'" target="_blank">`

Comment (NiLSPACE) [inline web_chat.lua]: While trying this I found that if you say `=` in the webadmin the webserver registers it as an empty string, and if you have something else in the message like `test=help` then the server (or plugin) won't receive anything at all.

Comment (NiLSPACE) [inline web_chat.lua]: I think I fixed it now.

**Comment by NiLSPACE** [comment]
Merge?

**Pull request merged**